C#/Botkit/Microsoft.BotKit/BotkitFrameworkAdapter.cs

```diff
@@ -1,33 +1,105 @@
         /// </summary>
-        /// <param name="turnContext">A TurnContext object representing a message or event from a user in Teams</param>
-        /// <returns>An array of channels in the format [{name: string, id: string}]</returns>
+        /// <param name="turnContext">A TurnContext object representing a message or event from a user in Teams.</param>
+        /// <returns>An array of channels in the format [{name: string, id: string}].</returns>
         public async Task<Tuple<string, string>> GetChannels(TurnContext turnContext)
         {
-            return new Tuple<string, string>("","");
+            if(turnContext.Activity.ChannelData != null && (turnContext.Activity.ChannelData as dynamic).team != null) //TO-DO: replace 'as dynamic'
+            {
+                var token = await _credentials.GetTokenAsync(true);
+
+                var uri = $"{ turnContext.Activity.ServiceUrl }v3/teams/{ (turnContext.Activity.ChannelData as dynamic).team.id }/conversations/"; //TO-DO: replace 'as dynamic'
+
+                return await RequestUrl(token, uri);
+            }
+            else
+            {
+                return new Tuple<string, string>("", "");
+            }
+        }
+
+        /// <summary>
+        /// Perform the Get request.
+        /// </summary>
+        /// <param name="token">An authorization token.</param>
+        /// <param name="uri">Url for the Get method.</param>
+        public async Task<Tuple<string, string>> RequestUrl(string token, string uri)
+        {
+            using (var client = new HttpClient())
+            {
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue($"Bearer { token }");
+
+                HttpResponseMessage response = await client.GetAsync(uri);
+
+                var result = await response.Content.ReadAsStringAsync();
+                var json = JsonConvert.DeserializeObject(result);
+                var conversations = (json as dynamic).conversations; //TO-DO: replace 'as dynamic'
+
+                if (conversations != null && conversations.count > 0)
+                {
+                    foreach(var c in conversations)
+                    {
+                        if(c.name is null)
+                        {
+                            c.name = "General";
+                        }
+                    }
+                    return conversations;
+                }
+                else
+                {
+                    return new Tuple<string, string>("", "");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Allows for mocking of the connector client in unit tests.
+        /// </summary>
+        /// <param name="serviceUrl">Clients service url.</param>
+        protected ConnectorClient CreateConnectorClient(string serviceUrl)
+        {
+            var client = new ConnectorClient(new Uri(serviceUrl), _credentials);
```

Comment:
Can we just `return new ConnectorClient` instead of using a variable here?

Reply:
Feedback applied. Thanks!